web-app/js/portal/ui/openlayers/layer/NcWMS.js

```diff
@@ -102,9 +102,7 @@ OpenLayers.Layer.NcWMS = OpenLayers.Class(OpenLayers.Layer.WMS, {
         });
     },
 
-    _timeSeriesDatesLoaded: function(datesWithData) {
```

Comment:
It took me some time to figure out what all the similarly named functions do in this class, listed here:

```
_fetchTimeSeriesForDay
_getTimeSeriesUrl
_loadTimeSeriesDates
_timeSeriesDatesLoaded
_timeSeriesLoadedForDate
getTimeSeriesForDay
loadTimeSeriesForDay
```

Firstly, I just wonder if this could be made a bit more understandable by doing the following:

* change `timeSeriesDates` simply to `Days`, e.g. `_loadTimeSeriesDates` becomes `loadDays`.  
* similarly, change `timeSeriesForDay` to `timesForDay`... 

This is because I think `timeSeries` is a bit redundant in the names, and `Date` is a bit ambiguous, in this context.

Secondly, correct me if I'm wrong here, but the general algorithm is now:

1. Fetch all the days that the layer covers
2. Fetch all the times on the *first* and the *last* day

Could this algorithm be more clearly expressed by bringing in a new class which sits between `NcWms` and `TemporalExtent`, and deals solely with this stuff?   It's just a bit confusing atm, IMO.